### PR TITLE
Validate types for Julia (part 1) 

### PR DESCRIPTION
**Pull request by axic**
Depends on #2310 (to add actual tests for the change).

### CODE REVIEW

**Comment by axic** [comment]
This is ready now.

File: libsolidity/inlineasm/AsmAnalysis.cpp [hunk 1]
```diff
@@ -453,3 +454,16 @@ Scope& AsmAnalyzer::scope(Block const* _block)
 	solAssert(scopePtr, "Scope requested but not present.");
 	return *scopePtr;
 }
+
+void AsmAnalyzer::expectValidType(string const& type, SourceLocation const& _location)
+{
+	if (!m_julia)
+		return;
+
+	if (!(set<string>{"bool", "u8", "s8", "u32", "s32", "u64", "s64", "u128", "s128", "u256", "s256"}).count(type))
```
Comment (chriseth) [inline libsolidity/inlineasm/AsmAnalysis.cpp]: Can you move the list of valid types to some more prominent position, e.g. a variable in an anonymous namespace in this cpp file?

File: libsolidity/inlineasm/AsmAnalysis.cpp [hunk 2]
```diff
@@ -453,3 +454,16 @@ Scope& AsmAnalyzer::scope(Block const* _block)
...
+		m_errors.push_back(make_shared<Error>(
+			Error::Type::TypeError,
+			"User defined types (\"" + type + "\") are not supported yet.",
```
Comment (chriseth) [inline libsolidity/inlineasm/AsmAnalysis.cpp]: Perhaps better `<type> is not a valid type (user defined types are not yet supported)`?

**Review by chriseth — CHANGES_REQUESTED** [review]

**Comment by axic** [comment]
@chriseth updated

File: libsolidity/inlineasm/AsmAnalysis.h [hunk 3]
```diff
@@ -61,11 +61,12 @@ struct AsmAnalysisInfo;
 class AsmAnalyzer: public boost::static_visitor<bool>
 {
 public:
-	AsmAnalyzer(
+	explicit AsmAnalyzer(
```
Comment (chriseth) [inline libsolidity/inlineasm/AsmAnalysis.h:64]: `explicit` is only useful if there is a constructor that takes only one (required) argument. But it does not hurt either.

**Review by chriseth — APPROVED** [review]

**Pull request merged**